src/Extension/FluentVersionedExtension.php

```diff
@@ -325,6 +325,83 @@ public function existsInLocale($locale = null)
         return $this->isDraftedInLocale($locale) || $this->isPublishedInLocale($locale);
     }
 
+    /**
+     * Check if this record has modifications in this locale
+     * Fluent friendly version of @see Versioned::stagesDiffer()
+     *
+     * @param string|null $locale
+     * @return bool
+     */
+    public function stagesDifferInLocale($locale = null)
+    {
+        $record = $this->owner;
+        $id = $record->ID ?: $record->OldID;
+        $class = get_class($record);
+
+        // Need to check if it's versioned
+        if (!$record->hasExtension(Versioned::class)) {
+            return false;
+        }
+
+        // Need to check that it has stages and is not new
+        if (!$id || !$record->hasStages()) {
+            return false;
+        }
+
+        // Need to check that it is actually localised
+        if (!$record->hasExtension(FluentExtension::class)) {
+            return false;
+        }
+
+        // Need to check that the localised version is versioned
+        if (!$record->hasExtension(FluentVersionedExtension::class)) {
```

Comment:
$record is the owner extended by FluentVersionedExtension; I think these are probably redundant.

note: If you move this to RecordLocale, then it's no longer redundant. :)